pkg/secretless/plugin/sharedobj/available_plugins_test.go

```diff
@@ -16,26 +19,111 @@ type mockTCPPlugin struct{ tcp.Plugin }
 // These need to sit outside of the func since we are comparing them in the
 // tests
 var mockHTTPPlugins = map[string]http.Plugin{
-	"one": mockHTTPPlugin{},
-	"two": mockHTTPPlugin{},
+	"one": &mockHTTPPlugin{},
+	"two": &mockHTTPPlugin{},
 }
 
 var mockTCPPlugins = map[string]tcp.Plugin{
-	"one":   mockTCPPlugin{},
-	"two":   mockTCPPlugin{},
-	"three": mockTCPPlugin{},
+	"one":   &mockTCPPlugin{},
+	"two":   &mockTCPPlugin{},
+	"three": &mockTCPPlugin{},
 }
 
-func getMockPlugins() plugin.AvailablePlugins {
+func mockPlugins() plugin.AvailablePlugins {
 	return &Plugins{
 		HTTPPluginsByID: mockHTTPPlugins,
 		TCPPluginsByID:  mockTCPPlugins,
 	}
 }
 
+func mockInternalPlugins() plugin.AvailablePlugins {
+	return &Plugins{
+		HTTPPluginsByID: map[string]http.Plugin{
+			"intHTTP1": &mockHTTPPlugin{},
+			"intHTTP2": &mockHTTPPlugin{},
+			"intHTTP3": &mockHTTPPlugin{},
+		},
+		TCPPluginsByID: map[string]tcp.Plugin{
+			"intTCP1": &mockTCPPlugin{},
+			"intTCP2": &mockTCPPlugin{},
+			"intTCP3": &mockTCPPlugin{},
+		},
+	}
+}
+
+func mockGetInternalPlugins() (plugin.AvailablePlugins, error) {
+	return mockInternalPlugins(), nil
+}
+
+func mockExternalPlugins() plugin.AvailablePlugins {
+	return &Plugins{
+		HTTPPluginsByID: map[string]http.Plugin{
+			"extHTTP1": &mockHTTPPlugin{},
+			"extHTTP2": &mockHTTPPlugin{},
+		},
+		TCPPluginsByID: map[string]tcp.Plugin{
+			"extTCP1": &mockTCPPlugin{},
+			"extTCP2": &mockTCPPlugin{},
+			"extTCP3": &mockTCPPlugin{},
+		},
+	}
+}
+
+func mockGetExternalPlugins(
+	pluginDir string,
+	checksumfile string,
+	logger log_api.Logger,
+) (plugin.AvailablePlugins, error) {
+
+	return mockExternalPlugins(), nil
+}
+
+func assertMapContainsHTTPPlugins(
+	t *testing.T,
+	allPlugins plugin.AvailablePlugins,
+	expectedPlugins map[string]http.Plugin,
+	comparePointers bool,
+) {
+
+	for name, httpPlugin := range expectedPlugins {
+		assert.Contains(t, allPlugins.HTTPPlugins(), name)
+		if _, ok := allPlugins.HTTPPlugins()[name]; !ok {
+			t.Fail()
+		}
+
+		if comparePointers {
+			assert.Equal(t, httpPlugin, allPlugins.HTTPPlugins()[name])
+		}
+	}
+
+}
+
+func assertMapContainsTCPPlugins(
+	t *testing.T,
+	allPlugins plugin.AvailablePlugins,
+	expectedPlugins map[string]tcp.Plugin,
+	comparePointers bool,
+) {
+
+	for name, tcpPlugin := range expectedPlugins {
+		assert.Contains(t, allPlugins.TCPPlugins(), name)
+		if _, ok := allPlugins.TCPPlugins()[name]; !ok {
+			t.Fail()
+		}
+
+		if comparePointers {
+			assert.Equal(t, tcpPlugin, allPlugins.TCPPlugins()[name])
+		}
+	}
+}
+
+func newLogger() log_api.Logger {
+	return log.New(true)
+}
+
```

Comment:
Def not a suggestion for now as it's not related to this code, but this makes me think we should have used the functional options patterns with a logger, so we could write:
```go
log.New(log.WithDebug, etc...)
```
The "true" it totally meaningless unless I'm viewing with my IDE.

Reply:
I agree but personally I would like use to move to a structured logging levels which would make that happen by the nature of it: 
```
log.New(etc...)
log.SetLevel(log.DebugSeverity)
```
 (maybe the invocation would be different though)